freud/plot.py

```diff
@@ -440,19 +441,25 @@ def voronoi_plot(box, polytopes, ax=None, color_by_sides=True, cmap=None):
         ax = fig.subplots()
 
     # Draw Voronoi polytopes
-    patches = [Polygon(poly[:, :2]) for poly in polytopes]
+    patches = [Polygon(poly[:, :2]) for poly in voronoi.polytopes]
     patch_collection = PatchCollection(patches, edgecolors="black", alpha=0.4)
 
-    if color_by_sides:
-        colors = np.array([len(poly) for poly in polytopes])
+    if color_by == "sides":
+        colors = np.array([len(poly) for poly in voronoi.polytopes])
         num_colors = np.ptp(colors) + 1
-    else:
+    elif color_by == "area":
+        colors = voronoi.volumes
+        num_colors = None  # creates a continuous colormap
+    elif color_by is None:
         colors = np.random.RandomState().permutation(np.arange(len(patches)))
         num_colors = np.unique(colors).size
+    else:
+        raise RuntimeError(f"Invalid color_by option {color_by}.")
 
     # Ensure we have enough colors to uniquely identify the cells
-    if cmap is None:
-        if color_by_sides and num_colors <= 10:
+    continuous_colorby_options = ["area"]
+    if cmap is None and color_by not in continuous_colorby_options:
```

Comment:
```suggestion
    if cmap is None and num_colors is not None:
```
This is equivalent and simpler.

Reply:
I wrote this in a way that would allow us to more easily add options for the `color_by` argument. When we want to add a new option, we can just add it to the proper list/dictionary. The same reasoning applies to the next 2 comments you left.

Reply:
Sure, but is that something we are really planning for? Being flexible is great but here it just adds code (in my opinion). There is also no guarantee that the next option, if there ever is one, would be continuous. I wouldn't try to block the PR for this though, so if you want to do it this way, then use `{"area"}` since a set is really what you want.

Reply:
If the next option is not continuous, then we would just not add it to the list named `continuous_colorby_options`, which intuitively makes sense. If the next option was continuous, then we would only add an element to a list and everything would just work properly. I chose to write it this way so it will be more easily extended for more options. Also, defining the list allows me an extra variable to explain what's going on, which is naturally more self-documenting than the suggested change.

The same logic holds for the other suggested changes below this one.